projects/plugins/jetpack/extensions/blocks/blogroll/utils.js

```diff
@@ -20,3 +20,28 @@ export function createBlockFromSubscription( subscription ) {
 		description,
 	} );
 }
+
+export function getValidDomain( siteURL ) {
+	if ( ! siteURL ) {
+		return null;
+	}
+
+	const pattern = new RegExp(
+		'^([a-zA-Z]+:\\/\\/)?' + // protocol
+			'((([a-z\\d]([a-z\\d-]*[a-z\\d])?)\\.)+[a-z]{2,})', // domain name
+		'i'
+	);
+
+	try {
+		return new URL( siteURL )?.host;
+	} catch ( e ) {
+		return siteURL.match( pattern ) ? siteURL.match( pattern )[ 2 ] : null;
```

Comment:
This runs the regex once:
```suggestion
		return siteURL.match( pattern )?.[ 2 ] ?? null;
```